lib/puppet/provider/cs_primitive/pcs.rb

```diff
@@ -260,7 +260,7 @@ def flush
       cmd += metadatas unless metadatas.nil?
       self.class.run_command_in_cib(cmd, @resource[:cib])
       if @property_hash[:promotable] == :true
-        cmd = [command(:pcs), 'resource', 'update', "ms_#{@property_hash[:name]}", (@property_hash[:name]).to_s]
```

Comment:
Is there a version of PCS for which the original code functioned? I was attempting to use the base code in my environment and it attempted a PCS command which failed. I couldn't find any examples of a pcs update command against a master resource which referenced the underlying type.  See issue #459 for additional detail.

Reply:
To be honest, no clue; I've only been using it on `crm` variants. I merely looked up if the original thing would be a valid command and it could work in both ways before and after the diff, and this diff *would* create at least another device *if* it has been working beforehand.

Reply:
@bastelfreak Not sure on your opinion on this here, if this really never worked I don't see something that would warrant a major version bump, but if it does work somehow it probably should.

Reply:
I'm not familiar with the crm variant, however, I believe the `pcs resource update` command can only target a single primitive. Here's the prototype from the [man page](
https://www.systutorials.com/docs/linux/man/8-pcs/):

```bash
pcs resource update <resource id> [resource options] [op [<operation action> <operation options>]...] [meta <meta operations>...] [--wait[=n]]
```

The original code was doing something more like:

```bash
pcs resource update <master resource id> <base resource id> [resource options] [meta ..]
```

I've done a fair bit of searching and I can't find any examples where this command structure would work.

Reply:
I'm thinking that the inspiration behind this syntax stems from at least [this RedHat documentation](https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux/6/html/configuring_the_red_hat_high_availability_add-on_with_pacemaker/s1-multistateresource-haar), which has:

```
pcs resource master master/slave_name resource_id|group_name [master_options]
```

as an alternate syntax. As I said, I haven't tested this in any way or form, but presence in the RHEL 6 documentation would imply it working with a RHEL 6.

Reply:
Ah, that would make sense. Do we really need to make a distinction about a breaking change for a OS version that the module documentation and metadata already explicitly states is unsupported?